src/day.rs

```diff
@@ -0,0 +1,231 @@
+use core::cmp::Ordering;
+use core::convert::TryFrom;
+use core::fmt;
+use core::fmt::Debug;
+use core::fmt::Display;
+use core::fmt::Write;
+use core::ops::Add;
+use core::ops::Sub;
+
+use crate::oldtime;
+use crate::DateTime;
+use crate::Days;
+use crate::NaiveDate;
+use crate::NaiveTime;
+use crate::TimeZone;
+use oldtime::Duration;
+
+/// Represents the full range of local timestamps in the day
+///
+///
+#[derive(Clone, Copy)]
+pub struct Day<Tz>
+where
+    Tz: TimeZone + Copy + Display,
+{
+    date: NaiveDate,
+    tz: Tz,
+}
+
+impl<Tz> Debug for Day<Tz>
+where
+    Tz: TimeZone + Copy + Display,
+{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        Debug::fmt(&self.date, f)?;
+        f.write_char(' ')?;
+        self.tz.fmt(f)
+    }
+}
+
+impl<Tz> Display for Day<Tz>
+where
+    Tz: TimeZone + Copy + Display,
+{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        Debug::fmt(self, f)
+    }
+}
+
+impl<Tz> PartialOrd for Day<Tz>
+where
+    Tz: TimeZone + Copy + Display,
+{
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        self.date.partial_cmp(&other.date)
+    }
+}
+
+impl<Tz> Ord for Day<Tz>
+where
+    Tz: TimeZone + Copy + Display,
+{
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.date.cmp(&other.date)
+    }
+}
+
+impl<Tz> PartialEq for Day<Tz>
+where
+    Tz: TimeZone + Copy + Display,
+{
+    fn eq(&self, other: &Self) -> bool {
+        self.date.eq(&other.date)
+    }
+}
+
+impl<Tz> Eq for Day<Tz> where Tz: TimeZone + Copy + Display {}
+
+impl<Tz> Day<Tz>
+where
+    Tz: TimeZone + Copy + Display,
+{
+    ///
+    pub fn date(&self) -> NaiveDate {
+        self.date
+    }
+
+    ///
+    pub fn zone(&self) -> Tz {
+        self.tz
+    }
+
+    ///
+    pub fn new(date: NaiveDate, tz: Tz) -> Day<Tz> {
+        Day { date, tz }
+    }
+
+    ///
+    pub fn succ(&self) -> Option<Day<Tz>> {
+        Some(Day { date: self.date.succ_opt()?, tz: self.tz })
+    }
+
+    ///
+    pub fn pred(&self) -> Option<Day<Tz>> {
+        Some(Day { date: self.date.pred_opt()?, tz: self.tz })
+    }
```

Comment:
As mentioned elsewhere, I kinda want to move away from providing these...

Reply:
I've removed, however these are potentially useful when dealing with non-instant style types, for example, `my_day.succ().start()` is an exclusive end time for `my_day` - but this could be implemented directly